test/utils.cpp

```diff
@@ -1045,3 +1045,112 @@ TEST_CASE("unescape_url() takes a percent-encoded string and returns the string
 	REQUIRE(utils::unescape_url("%00") == "");
 
 }
+
+TEST_CASE("mkdir_parents() creates all paths components and returns 0 if "
+		"the path now exists",
+		"[utils]")
+{
+	TestHelpers::TempDir tmp;
+
+	const auto require_return_zero = [](const std::string& path) {
+		REQUIRE(utils::mkdir_parents(path.c_str(), 0700) == 0);
+		REQUIRE(::access(path.c_str(), R_OK | X_OK) == 0);
+	};
+
+	SECTION("Simple test on temporary dir itself") {
+		const auto path = tmp.getPath();
+		INFO("Path is " << path);
+		require_return_zero(path);
+	}
+
+	SECTION("Zero intermediate directories") {
+		const auto path = tmp.getPath() + std::to_string(rand());
+		INFO("Path is " << path);
+
+		SECTION("Target doesn't yet exist") {
+			require_return_zero(path);
+		}
+
+		SECTION("Target already exists") {
+			REQUIRE(::mkdir(path.c_str(), 0700) == 0);
+			require_return_zero(path);
+		}
+	}
+
+	SECTION("One intermediate directory") {
+		const auto intermediate_path = tmp.getPath() + std::to_string(rand());
+		const auto path = intermediate_path + "/" + std::to_string(rand());
+		INFO("Path is " << path);
+
+		SECTION("Which doesn't exist") {
+			require_return_zero(path);
+		}
+
+		SECTION("Which exists") {
+			REQUIRE(::mkdir(intermediate_path.c_str(), 0700) == 0);
```

Comment:
It doesn't exist, because [Catch executes one leaf section per TEST_CASE run](https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#test-cases-and-sections), e.g.:

```c++
TEST_CASE("example") {
    a();
    SECTION("one") { b(); }
    SECTION("two") { c(); }
    SECTION("three") {
        d();
        SECTION("four") { e(); }
    }
}
```

results in three runs:
- `a()`, `b()`;
- `a()`, `c()`;
- `a()`, `d()`, `e()`.

Reply:
Thank you for the link to catch's documentation, so please ignore my comment there. Looks fine to me in this case.